lib/pagination/adapter.js

```diff
@@ -19,10 +19,21 @@ const totalPages = adapterParams => Math.ceil(totalCount(adapterParams) / adapte
 const nextPage = adapterParams =>
   adapterParams.page >= totalPages(adapterParams) ? null : adapterParams.page + 1;
 
-const pageUrl = (adapterParams, pageParam) =>
-  pageParam
-    ? `http://${adapterParams.request.headers.host}${adapterParams.request.url}?page=${pageParam}`
-    : null;
+const trimPageUrl = (originalUrl, newPageNum) => {
+  const newPageUrl = originalUrl.replace(/page=./, `page=${newPageNum}`);
```

Comment:
Is this taking into account that the page might be a 2 or more digits number ? Because I think that `.` will just get 1 digit and if you are in page `10` for example you will only replace `page=1`

Reply:
You're totally right! I missed that posibility. Will cover that in the next commit